### PR TITLE
chore(docs): update Helm Chart readme with Azure KeyVault info

### PR DESCRIPTION
**Pull request by migmartri**
Update Helm Chart readme file with instructions on how to configure Azure Key Vault at deployment time.

Refs #376

### CODE REVIEW

File: deployment/chainloop/README.md [hunk 1]
```diff
@@ -311,7 +327,7 @@ controlplane:
 
 ### Use AWS secret manager
```
Comment (danlishka) [inline deployment/chainloop/README.md:328]: AWS Secrets Manager - this is the name https://docs.aws.amazon.com/secretsmanager/latest/userguide/intro.html

File: deployment/chainloop/README.md [hunk 2]
```diff
@@ -311,7 +327,7 @@ controlplane:
...
 
-You can swap the secret manager backend with the following settings
+Instead of using [Hashicorp Vault](https://www.vaultproject.io/) (default), you can use [AWS Secret Manager](https://aws.amazon.com/secrets-manager/) by adding these settings in your `values.yaml` file
```
Comment (danlishka) [inline deployment/chainloop/README.md:330]: Secret -> Secrets

File: deployment/chainloop/README.md [hunk 3]
```diff
@@ -334,6 +350,21 @@ secretsBackend:
     serviceAccountKey: [KEY]
 ```
 
+### Use Azure KeyVault as credentials backend
```
Comment (danlishka) [inline deployment/chainloop/README.md:353]: Why don't you follow the convention from previous sections?

I would either say "Use Azure secrets manager" or "Use Azure KeyVault"

File: devel/README.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ To get up and running you'll need
 Chainloop requires some configuration to be in place such as
 
 - PostgreSQL 14 connection from the Control plane
-- Access to a secrets manager. Currently Hashicorp Vault and AWS secret manager are supported.
+- Access to a secrets manager. Currently Hashicorp Vault, AWS secret manager, GCP secret manager and Azure KeyVault are supported.
```
Comment (danlishka) [inline devel/README.md:22]: Currently Hashicorp Vault, AWS Secrets Manager, Google Cloud Secret Manager, and Azure KeyVault.

**Review by danlishka — APPROVED** [review]
LGTM!

File: deployment/chainloop/README.md (hunk 1 above)
Comment (migmartri) [reply deployment/chainloop/README.md:328]: Good catch!

**Pull request merged**